### PR TITLE
[ts-sdk] Fix default for coin client transfer and register on default

### PR DESCRIPTION
**Pull request by gregnazario**
### Description
Change default in the SDK to transfer with register.  Can still be turned off by specifying false

### CODE REVIEW

File: ecosystem/typescript/sdk/src/plugins/coin_client.ts [hunk 1]
```diff
@@ -87,7 +87,12 @@ export class CoinClient {
 
     // If we should create the receiver account if it doesn't exist on-chain,
     // use the `0x1::aptos_account::transfer` function.
-    const func = extraArgs?.createReceiverIfMissing ? "0x1::aptos_account::transfer_coins" : "0x1::coin::transfer";
+    let func: string;
+    if (extraArgs?.createReceiverIfMissing === undefined) {
```
Comment (0xmaayan) [inline ecosystem/typescript/sdk/src/plugins/coin_client.ts:91]: do we have tests to verify this logic?

**Review by banool — COMMENTED** [review]
This is a breaking change, can we mark it as such? Also we should make it possible to opt in to the old behaviour still for those who know they don't need the new function.

**Comment by gregnazario** [comment]
> This is a breaking change, can we mark it as such? Also we should make it possible to opt in to the old behaviour still for those who know they don't need the new function.

You can always provide `false` to use the old default.

**Comment by 0xmaayan** [comment]
> > This is a breaking change, can we mark it as such? Also we should make it possible to opt in to the old behaviour still for those who know they don't need the new function.
> 
> You can always provide `false` to use the old default.

we can have it `false` by default

**Comment by gregnazario** [comment]
> > > This is a breaking change, can we mark it as such? Also we should make it possible to opt in to the old behaviour still for those who know they don't need the new function.
> > 
> > 
> > You can always provide `false` to use the old default.
> 
> we can have it `false` by default

The previous default was `false` because of `falsyness`

**Review by davidiw — APPROVED** [review]

File: ecosystem/typescript/sdk/src/plugins/coin_client.ts (hunk 1 above)
Comment (gregnazario) [reply ecosystem/typescript/sdk/src/plugins/coin_client.ts:93]: Test added

**Pull request merged**